pkg/lib/search/pgsearch/service.go

```diff
@@ -1,20 +1,110 @@
 package pgsearch
 
 import (
+	"context"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/lib/pq"
+
 	apimodel "github.com/authgear/authgear-server/pkg/api/model"
 	"github.com/authgear/authgear-server/pkg/lib/authn/user"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/infra/db"
 	"github.com/authgear/authgear-server/pkg/lib/infra/db/searchdb"
 	"github.com/authgear/authgear-server/pkg/util/graphqlutil"
 )
 
 type Service struct {
+	AppID       *config.AppID
 	SQLBuilder  *searchdb.SQLBuilder
 	SQLExecutor *searchdb.SQLExecutor
 }
 
 func (s *Service) QueryUser(
+	ctx context.Context,
 	searchKeyword string,
 	sortOption user.SortOption,
 	pageArgs graphqlutil.PageArgs) ([]apimodel.PageItemRef, error) {
-	return nil, nil
+	if s.SQLExecutor == nil {
+		return nil, fmt.Errorf("search database credential is not provided")
+	}
+	var refs []apimodel.PageItemRef
+	err := s.SQLExecutor.Database.ReadOnly(ctx, func(ctx context.Context) error {
```

Comment:
Inject the handle directly. Do not use the field of a dependency.

Reply:
Fixed in later commits